verification/lid_driven_cavity/lid_driven_cavity.jl

```diff
@@ -0,0 +1,97 @@
+using Printf
+using Logging
+
+using Oceananigans
+using Oceananigans: Face, Cell
+using Oceananigans.Diagnostics
+using Oceananigans.OutputWriters
+using Oceananigans.AbstractOperations
+
+Logging.global_logger(OceananigansLogger())
+
+function simulate_lid_driven_cavity(; Re, N, end_time)
+    topology = (Flat, Bounded, Bounded)
+    domain = (x=(0, 1), y=(0, 1), z=(0, 1))
+    grid = RegularCartesianGrid(topology=topology, size=(1, N, N); domain...)
+
+    v_bcs = VVelocityBoundaryConditions(grid,
+           top = ValueBoundaryCondition(1.0),
```

Comment:
Trivial side note: I think it's perfectly fine to use `1` instead of `1.0` and I do not think there is any effect on performance (since type promotion is cheap --- correct me if I'm wrong!)

Reply:
Ah good point, the type promotion only happens once.